Working on this error https://github.com/googleapis/google-cloud-java/pull/12831/checks?check_run_id=71851340776:

```
F
======================================================================
FAIL: test_monorepo_generation (integration_tests.IntegrationTest.test_monorepo_generation)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/workspace/sdk-platform-java/.cloudbuild/library_generation/scripts/integration_tests.py", line 91, in test_monorepo_generation
    self.assertTrue(
AssertionError: False is not true :   The generated java-apigee-connect/.repo-metadata.json is different from golden.
----------------------------------------------------------------------
Ran 1 test in 0.021s
FAILED (failures=1)
**************************************************
Checking for differences in 'java-apigee-connect'.
  The expected library is in /workspace/golden/java-apigee-connect.
  The actual library is in /workspace/google-cloud-java/java-apigee-connect.
  No differences found in java-apigee-connect
```

--- a/sdk-platform-java/hermetic_build/library_generation/tests/resources/test_monorepo_postprocessing/java-dns/pom.xml
+++ b/sdk-platform-java/hermetic_build/library_generation/tests/resources/test_monorepo_postprocessing/java-dns/pom.xml
@@ -4,6 +4,6 @@
   <groupId>com.google.cloud</groupId>
   <artifactId>google-cloud-dns</artifactId>
   <packaging>jar</packaging>
-  <version>2.89.0</version><!-- {x-version-update:google-cloud-dns:current} -->
+  <version>2.33.0-SNAPSHOT</version><!-- {x-version-update:google-cloud-dns:current} -->
   <name>Google Cloud DNS Parent</name>
 </project>

--- a/sdk-platform-java/hermetic_build/library_generation/tests/resources/test_monorepo_postprocessing/java-service-control/google-cloud-service-control-bom/pom.xml
+++ b/sdk-platform-java/hermetic_build/library_generation/tests/resources/test_monorepo_postprocessing/java-service-control/google-cloud-service-control-bom/pom.xml
@@ -3,6 +3,6 @@
   <modelVersion>4.0.0</modelVersion>
   <groupId>com.google.cloud</groupId>
   <artifactId>google-cloud-service-control-bom</artifactId>
-  <version>1.91.0</version><!-- {x-version-update:google-cloud-service-control:current} -->
+  <version>1.35.0-SNAPSHOT</version><!-- {x-version-update:google-cloud-service-control:current} -->
   <packaging>pom</packaging>
 </project>

--- a/sdk-platform-java/hermetic_build/library_generation/tests/resources/test_monorepo_postprocessing/java-tasks/google-cloud-tasks-bom/pom.xml
+++ b/sdk-platform-java/hermetic_build/library_generation/tests/resources/test_monorepo_postprocessing/java-tasks/google-cloud-tasks-bom/pom.xml
@@ -3,6 +3,6 @@
   <modelVersion>4.0.0</modelVersion>
   <groupId>com.google.cloud</groupId>
   <artifactId>google-cloud-tasks-bom</artifactId>
-  <version>2.91.0</version><!-- {x-version-update:google-cloud-tasks:current} -->
+  <version>2.35.0-SNAPSHOT</version><!-- {x-version-update:google-cloud-tasks:current} -->
   <packaging>pom</packaging>
 </project>

--- a/sdk-platform-java/hermetic_build/library_generation/utils/utilities.py
+++ b/sdk-platform-java/hermetic_build/library_generation/utils/utilities.py
@@ -251,7 +251,10 @@ def generate_postprocessing_prerequisite_files(
         "distribution_name": library.get_maven_coordinate(),
     }
 
-    if library.get_library_name() not in LIBRARIES_WITHOUT_API_ID and library.library_type != "OTHER":
+    if (
+        library.get_library_name() not in LIBRARIES_WITHOUT_API_ID
+        and library.library_type != "OTHER"
+    ):
         repo_metadata["api_id"] = api_id
 
     repo_metadata["library_type"] = library.library_type

--- a/sdk-platform-java/hermetic_build/library_generation/utils/utilities.py
+++ b/sdk-platform-java/hermetic_build/library_generation/utils/utilities.py
@@ -26,7 +26,16 @@ from library_generation.utils.file_render import render
 from library_generation.utils.proto_path_utils import remove_version_from
 
 script_dir = os.path.dirname(os.path.realpath(__file__))
-SDK_PLATFORM_JAVA = "googleapis/google-cloud-java"
+
+LIBRARIES_WITHOUT_API_ID = {
+    "google-auth-library",
+    "showcase",
+    "iam",
+    "api-common",
+    "common-protos",
+    "gax",
+    "core",
+}
 
 
 def create_argument(arg_key: str, arg_container: object) -> List[str]:
@@ -179,9 +188,7 @@ def get_library_repository(
 
     :return: string representing the repository
     """
-    if config.contains_common_protos():
-        repo = SDK_PLATFORM_JAVA
-    elif config.is_monorepo():
+    if config.is_monorepo() or config.contains_common_protos():
         repo = "googleapis/google-cloud-java"
     else:
         repo = f"googleapis/{language}-{library.get_library_name()}"
@@ -242,15 +249,13 @@ def generate_postprocessing_prerequisite_files(
         "repo": repo,
         "repo_short": f"{language}-{library_name}",
         "distribution_name": library.get_maven_coordinate(),
-        "api_id": api_id,
-        "library_type": library.library_type,
-        "requires_billing": library.requires_billing,
     }
 
-    # this removal is for java-common-protos and java-iam in
-    # sdk-platform-java
-    if repo == SDK_PLATFORM_JAVA:
-        repo_metadata.pop("api_id")
+    if library.get_library_name() not in LIBRARIES_WITHOUT_API_ID and library.library_type != "OTHER":
+        repo_metadata["api_id"] = api_id
+
+    repo_metadata["library_type"] = library.library_type
+    repo_metadata["requires_billing"] = library.requires_billing
 
     if library.api_reference:
         repo_metadata["api_reference"] = library.api_reference

